cmd/doctor.go

```diff
@@ -552,14 +552,19 @@ func runDoctorCheckDBConsistency(ctx *cli.Context) ([]string, error) {
 		}
 	}
 
+	var (
+		rPullRequest string = models.RealTableName("pull_request")
```

Comment:
This will not work. We should define all the table names and set them after settings loaded.

Reply:
@lunny Do you means this style? Thanks.
```DIFF
diff --git a/models/models.go b/models/models.go
index 684c1f290..2230d0a96 100644
--- a/models/models.go
+++ b/models/models.go
@@ -162,6 +162,7 @@ func NewTestEngine() (err error) {
 	if len(setting.Database.TableNamePrefix) > 0 {
 		x.SetTableMapper(names.NewPrefixMapper(x.GetTableMapper(), setting.Database.TableNamePrefix))
 	}
+	InitRealTableNameList()
 
 	x.SetLogger(NewXORMLogger(!setting.ProdMode))
 	x.ShowSQL(!setting.ProdMode)
@@ -179,6 +180,7 @@ func SetEngine() (err error) {
 	if len(setting.Database.TableNamePrefix) > 0 {
 		x.SetTableMapper(names.NewPrefixMapper(x.GetTableMapper(), setting.Database.TableNamePrefix))
 	}
+	InitRealTableNameList()
 	// WARNING: for serv command, MUST remove the output to os.stdout,
 	// so use log file to instead print to stdout.
 	x.SetLogger(NewXORMLogger(setting.Database.LogSQL))
@@ -324,3 +326,14 @@ func FindByMaxID(maxID int64, limit int, results interface{}) error {
 func RealTableName(tableName string) string {
 	return setting.Database.TableNamePrefix + tableName
 }
+
+var (
+	rUser = "`user`"
+	// ....
+)
+
+// InitRealTableNameList Init realtable name list
+func InitRealTableNameList() {
+	rUser = "`" + RealTableName("user") + "`"
+	// ...
+}
```

···

Reply:
Yes, that's what I mean.